bundles/org.openhab.binding.pentair/src/main/java/org/openhab/binding/pentair/internal/handler/PentairControllerHandler.java

```diff
@@ -0,0 +1,1134 @@
+                int day = p.getByte(3);
+                int month = p.getByte(4);
+                int year = p.getByte(5);
+
+                logger.debug("System Clock: {}:{} {} {}/{}/{}", hour, minute, dow, day, month, year);
+
+                break;
+            case 6: // Set run mode
+                // No action - have not verified these commands, here for documentation purposes and future enhancement
+                logger.debug("Set run mode {}: {}", p.getDest(), p.getByte(0));
+
+                break;
+            case 7: // Pump Status - handled in IntelliFlo handler
+                // No action - have not verified these commands, here for documentation purposes and future enhancement
+                logger.debug("Pump request status (unseen): {}", p);
+                break;
+            case 8: // Heat Status - A5 01 0F 10 08 0D 4B 4B 4D 55 5E 07 00 00 58 00 00 00 
+                if (p.getLength() != 0x0D) {
+                    logger.debug("Expected length of 13: {}", p);
+                    return;
+                }
+
+                PentairHeatStatus heatStatus = new PentairHeatStatus(p);
+                heatStatusCache.putValue(heatStatus);
+
+                refreshHeatStatusChannels(null, heatStatus);
+
+                logger.debug("Heat status: {}, {}, {}", p, heatStatus.poolSetPoint, heatStatus.spaSetPoint);
+                break;
+            case 10: // Custom Names
+                logger.debug("Get Custom Names (unseen): {}", p);
+                break;
+            case 11: // Circuit Names
+                int index;
+
+                index = p.getByte(0);
+                index--; // zero index
+                if (index < 0 || index >= NUM_CIRCUITS) {
+                    break;
+                }
+
+                PentairControllerCircuit circuit = new PentairControllerCircuit(index + 1, CIRCUIT_GROUPS[index]);
+
+                circuit.setName(p.getByte(2));
+                circuit.setFunction(p.getByte(1));
+
+                refreshCircuitChannels(null, circuit);
+
+                circuitsCache[index].putValue(circuit);
+
+                logger.debug("Circuit Names - Circuit: {}, Function: {}, Name: {}", circuit.id,
+                        circuit.circuitFunction.getFriendlyName(), circuit.circuitName.getFriendlyName());
+                break;
+            case 17: // schedule - A5 1E 0F 10 11 07 01 06 0B 00 0F 00 7F
+                int id;
+
+                id = p.getByte(PentairControllerSchedule.ID);
+                if (id < 1 || id > NUM_SCHEDULES) {
+                    break;
+                }
+
+                PentairControllerSchedule schedule = new PentairControllerSchedule();
+
+                schedule.parsePacket(p);
+
+                String groupID = schedule.getGroupID();
+
+                logger.debug("Controller schedule group: {}", groupID);
+
+                refreshScheduleChannels(null, schedule);
+
+                schedulesCache[id - 1].putValue(schedule);
+
```

Comment:
There are a lot of blank lines here that is making it harder to read. Please remove some of them.